libraries/platforms/dromajo-vcs/bsg_manycore_platform.cpp

```diff
@@ -0,0 +1,392 @@
+}
+
+/**
+ * Get the current cycle counter of the Manycore Platform
+ * @param[in]  mc     A manycore instance initialized with hb_mc_manycore_init()
+ * @param[out] time   The current counter value.
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_get_cycle(hb_mc_manycore_t *mc, uint64_t *time) {
+  return HB_MC_NOIMPL;
+}
+
+/**
+ * Get the number of instructions executed for a certain class of instructions
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @param[in] itype An enum defining the class of instructions to query.
+ * @param[out] count The number of instructions executed in the queried class.
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_get_icount(hb_mc_manycore_t *mc, bsg_instr_type_e itype, int *count) {
+  return HB_MC_NOIMPL;
+}
+
+/**
+ * Enable trace file generation (vanilla_operation_trace.csv)
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_trace_enable(hb_mc_manycore_t *mc) {
+  return HB_MC_NOIMPL;
+}
+
+/**
+ * Disable trace file generation (vanilla_operation_trace.csv)
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_trace_disable(hb_mc_manycore_t *mc) {
+  return HB_MC_NOIMPL;
+}
+
+/**
+ * Enable log file generation (vanilla.log)
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_log_enable(hb_mc_manycore_t *mc) {
+  return HB_MC_NOIMPL;
+}
+
+/**
+ * Disable log file generation (vanilla.log)
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */
+int hb_mc_platform_log_disable(hb_mc_manycore_t *mc) {
+  return HB_MC_NOIMPL;
+}
+
+/**
+ * Block until chip reset has completed.
+ * @param[in] mc    A manycore instance initialized with hb_mc_manycore_init()
+ * @return HB_MC_SUCCESS on success. Otherwise an error code defined in bsg_manycore_errno.h.
+ */        
+int hb_mc_platform_wait_reset_done(hb_mc_manycore_t *mc) {
+  hb_mc_packet_t reset_req_pkt, reset_resp_pkt;
+  int err;
+  uint32_t data;
+
+  reset_req_pkt.request.x_dst = (0 << 4) | 0;
+  reset_req_pkt.request.y_dst = (1 << 3) | 0;
+  reset_req_pkt.request.x_src = (0 << 4) | 0;
+  reset_req_pkt.request.y_src = (1 << 3) | 1;
+  reset_req_pkt.request.op_v2 = HB_MC_PACKET_OP_REMOTE_LOAD;
+  reset_req_pkt.request.payload = 0;
+  reset_req_pkt.request.reg_id = 0;
+  reset_req_pkt.request.addr = MC_RESET_DONE_EPA_ADDR;
+
```

Comment:
As dan commented too, you should replace magic numbers with macros

Reply:
Added a HOST_X_COORD and HOST_Y_COORD. I will get rid of the other magic numbers soon.